pkgs/tools/misc/pywb/default.nix

```diff
@@ -0,0 +1,92 @@
+{ lib, fetchFromGitHub, fetchPypi, python3, stdenv }:
+
+python3.pkgs.buildPythonApplication rec {
+  name = "pywb";
+  version = "2.7.4";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "webrecorder";
+    repo = "pywb";
+    rev = "v-${version}";
+    sha256 = "sha256-M44TwBzXuF33HNLB2OjmBKNDG9kt/qhGEOu589YGvsQ=";
+    # TODO: We clone the wombat submodule, but we still need to implement
+    # building wombat.
+    fetchSubmodules = true;
+  };
+
+  patches = [ ./pywb.patch ];
+
+  postPatch = ''
+    substituteInPlace requirements.txt \
+      --replace "jinja2<3.0.0" "jinja2" \
+      --replace "redis<3.0" "redis" \
+      --replace "markupsafe<2.1.0" "markupsafe" \
+      --replace "fakeredis<1.0" "fakeredis" \
+      --replace "gevent==21.12.0" "gevent"
+  '';
+
+  propagatedBuildInputs = with python3.pkgs; [
```

Comment:
These don't seem to match [upstream](https://github.com/webrecorder/pywb/blob/2.7.0/requirements.txt). `pytest` for example should usually not be propagated. Many maintainers in nixpkgs prefer to sort the propagated inputs the same way as upstream, making it easier to spot differences. If any packages have been added that were not referenced in upstream then

Reply:
[Extras](https://github.com/webrecorder/pywb/blob/2.7.0/setup.py#L110-L115) is commonly listed in [`passthru.optional-dependencies`](https://nixos.org/manual/nixpkgs/stable/#python-optional-dependencies)

Reply:
bump, please don't propagate pytest